spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/config/DiscoveryClientOptionalArgsConfiguration.java

```diff
@@ -174,19 +171,27 @@ public WebClientNotFoundConfiguration() {
 
 	}
 
-	static class JerseyClientPresentAndEnabledCondition extends AllNestedConditions {
+	static class RestTemplateEnabledCondition extends AnyNestedCondition {
 
-		JerseyClientPresentAndEnabledCondition() {
+		RestTemplateEnabledCondition() {
 			super(ConfigurationPhase.REGISTER_BEAN);
 		}
 
 		@ConditionalOnClass(name = "org.glassfish.jersey.client.JerseyClient")
-		static class OnJerseyClientPresent {
```

Comment:
The Jersey conditions are not related to the `WebClient`/ `RestClient`/ `RestTemplate` selection. They should be separate.

Reply:
Yes, I edited this incorrectly 👍